File: day1/rooks/public/py/grader.py
import rooks
import sys

def main():

    line_nmk = sys.stdin.readline().split()
    assert len(line_nmk) == 3
    N, M, K = map(int, line_nmk)
    
    A = []
    for i in range(N):
        row = list(map(int, sys.stdin.readline().split()))
        assert len(row) == M
        A.append(row)

    moves = rooks.calculate_moves(A, K)

    assert isinstance(moves, list)
    assert len(moves) == N
    
    
    for i in range(N):
        assert isinstance(moves[i], list)
        assert len(moves[i]) == M
        print(*moves[i])

if __name__ == "__main__":
    main()

File: day1/rooks/public/cpp/grader.cpp
#include "rooks.h"
#include <cassert>
#include <cstdio>
#include <vector>

int main() {
    int N, M, K;
    assert(scanf("%d %d %d", &N, &M, &K) == 3);
    std::vector<std::vector<int>> A(N, std::vector<int>(M));
    for (int i=0; i<N; i++) {
        for (int j=0; j<M; j++) assert(scanf("%d", &A[i][j]) == 1);
    }
    std::vector<std::vector<int>> moves = calculate_moves(A, K);
    assert(moves.size() == N);
    for (int i=0; i<N; i++) {
        assert(moves[i].size() == M);
        for (int j=0; j<M; j++) printf("%d ", moves[i][j]);
        printf("\n");
    }
    return 0;
}
